tor/controller.go

```diff
@@ -373,23 +385,25 @@ func (c *Controller) readResponse(expected int) (int, string, error) {
 func parseTorReply(reply string) map[string]string {
 	params := make(map[string]string)
 
-	// Replies can either span single or multiple lines, so we'll default
-	// to stripping whitespace and newlines in order to retrieve the
-	// individual contents of it. The -1 indicates that we want this to span
-	// across all instances of a newline.
-	contents := strings.Split(strings.Replace(reply, "\n", " ", -1), " ")
+	// Find all fields of a reply. The -1 indicates that we want this to
+	// find all instances of the regexp.
+	contents := replyFieldRegexp.FindAllString(reply, -1)
 	for _, content := range contents {
 		// Each parameter within the reply should be of the form
-		// "KEY=VALUE". If the parameter doesn't contain "=", then we
-		// can assume it does not provide any other relevant information
-		// already known.
+		// KEY=VALUE or KEY="VALUE".
 		keyValue := strings.SplitN(content, "=", 2)
-		if len(keyValue) != 2 {
-			continue
-		}
-
 		key := keyValue[0]
 		value := keyValue[1]
+
+		// If it's a quoted value, remove quotes and backslashes before
+		// escaped characters. A backslash followed by any character
+		// represents that character.
+		if strings.HasPrefix(value, `"`) {
+			value = strings.ReplaceAll(
+				value[1:len(value)-1], `\`, ``,
```

Comment:
nit: since we're only replacing backslashes, can't we simply do `value = strings.ReplaceAll(value, ...)` instead of the substring starting at index 1?

Reply:
It also removes the quotes!

Reply:
I intentionally wrote the regular expression in such a way that it finds each key=value pair as a single match. Then they can be split by `=`. If it removed the quotes, this code wouldn't run, no?

Reply:
Ah of course, you're totally right. Long day of code review... I was confused by the `HasPrefix` with `"` and then the quote not appearing in the `ReplaceAll`. But you're just doing two steps in one. Could perhaps write it his way (unnecessary perhaps but increases readability):

```
if strings.HasPrefix(value, `"`) {
	// Remove quotes around value.
	value = value[1:len(value)-1]

	// Remove backslash.
	value = unescapeValue(value)
```

Reply:
Done.